ooniprobe/Model/Database/Measurement.mm

```diff
@@ -96,22 +96,30 @@ -(void)deleteObject{
     [self remove];
 }
 
-- (void)getExplorerUrl:(void (^)(NSString *))measurement_url {
-    NSString *path = [NSString stringWithFormat:@"https://api.ooni.io/api/v1/measurements?report_id=%@&input=%@", self.report_id, self.url_id.url];
+-(void)getExplorerUrlSuccessCallback:(void (^)(NSString*))success errorCallback:(void (^)(NSError*))error {
+    NSLog(@"%@ getExplorerUrl",self.Id);
+    NSMutableString *path = [NSMutableString stringWithFormat:@"https://api.ooni.io/api/v1/measurements?report_id=%@",
+                      self.report_id];
+    if ([self.test_name isEqualToString:@"web_connectivity"])
+        [path appendString:[NSString stringWithFormat:@"&input=%@",
+                            self.url_id.url]];
```

Comment:
I would suggest here to see whether there is a Cocoa facility for building URLs in a robust way. I am especially concerned with proper escaping. I bet there is a safe Cocoa way to construct a query string and then attaching it to a URL in a way that escaping is on the library rather than on us.